aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -243,6 +247,160 @@ std::tuple<Tensor, Tensor> prelu_backward_cuda(const Tensor& grad_out_, const Te
   return std::tuple<Tensor, Tensor>{input_grad, weight_grad};
 }
 
+// -----------------------------------
+// rrelu
+// -----------------------------------
+template<typename scalar_t>
+inline scalar_t __device__ curand_uniform_type(curandStatePhilox4_32_10_t *state);
+
+template <>
+inline THHalf __device__ curand_uniform_type<THHalf>(curandStatePhilox4_32_10_t *state) {
+  auto rand = curand_uniform4(state);
+  return ScalarConvert<float, THHalf>::to(rand.x);
```

Comment:
using only .x out of 4 generated numbers is wasteful, you can have an unroll loop in the kernel that would use all the values, you can take a look e.g. at the non-vectorized `fused_dropout_kernel` in Dropout.cu